oqupy/tempo.py

```diff
@@ -876,13 +907,14 @@ def guess_tempo_parameters(
             break
 
     dt = np.min(times[1:] - times[:-1])
+    tcut = float(times[-1])
     dkmax = len(times)
     epsrel = _estimate_epsrel(dkmax, tolerance)
     sys.stderr.flush()
 
     return TempoParameters(
         dt=dt,
-        dkmax=dkmax,
+        tcut=tcut,
```

Comment:
Tempo actually cares about the integer dkmax. Given that TempoParameters can now again take either dkmax or tcut, I'd suggest to revert this particular change back to dkmax. Otherwise the integer (dkmax) gets converted into a float (tcut) and than back again (dkmax) in the TempoParameters parsing.

Reply:
Thanks, forgot I made that change already! I'll revert in next iteration.